common/src/main/java/net/william278/huskchat/message/ChatMessage.java

```diff
@@ -112,6 +113,11 @@ public void dispatch() {
                                     continue;
                                 }
                                 if (!sender.hasPermission("huskchat.command.localspy")) {
+                                    try {
+                                        PlayerCache.removeLocalSpy(sender);
+                                    } catch(IOException e) {
+                                        implementor.getLoggingAdapter().log(Level.SEVERE, "Failed to remove local spy after failed permission check");
```

Comment:
Same as with social spy, small snag - but don't forget to log the exception too.
`implementor.getLoggingAdapter().log(Level.SEVERE, "Failed to remove local spy after failed permission check", e);`